### PR TITLE
8315684: Parallelize sun/security/util/math/TestIntegerModuloP.java

### PR DESCRIPTION
**Pull request by ianrichr**
sun/security/util/math/TestIntegerModuloP.java runs in tier2 and takes about 600 seconds to run. Thus, it drags the execution time of tier2 on large machines. We can split the run configurations a bit so that test is more parallel. 

TestIntegerModuloP.java current run time: **235.02s user 6.60s system 119% cpu 3:22.69 total**
TestIntegerModuloP.java parallelized run time: **328.75s user 14.57s system 755% cpu 45.467 total**

This change splits TestIntegerModuloP.java's previously serialized test into 11 separate tests that run in parallel.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issue
 * [JDK-8315684](https://bugs.openjdk.org/browse/JDK-8315684): Parallelize sun/security/util/math/TestIntegerModuloP.java (**Enhancement** - P4)

### Reviewers
 * [Aleksey Shipilev](https://openjdk.org/census#shade) (@shipilev - **Reviewer**)
 * [Weijun Wang](https://openjdk.org/census#weijun) (@wangweij - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15618/head:pull/15618` \
`$ git checkout pull/15618`

Update a local copy of the PR: \
`$ git checkout pull/15618` \
`$ git pull https://git.openjdk.org/jdk.git pull/15618/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15618`

View PR using the GUI difftool: \
`$ git pr show -t 15618`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15618.diff">https://git.openjdk.org/jdk/pull/15618.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15618#issuecomment-1718332106)

### CODE REVIEW

File: test/jdk/sun/security/util/math/TestIntegerModuloP.java [hunk 1]
```diff
@@ -22,24 +22,85 @@
...
+ */
+
+ /*
+ * @test id=Curve448OrderField
+ * @modules java.base/sun.security.util java.base/sun.security.util.math java.base/sun.security.util.math.intpoly
+ * @build BigIntegerModuloP
  * @run main TestIntegerModuloP sun.security.util.math.intpoly.Curve448OrderField 56 12
  */
 
+
```
Comment (shipilev) [inline test/jdk/sun/security/util/math/TestIntegerModuloP.java:103]: Excess new line.

**Review by shipilev — COMMENTED** [review]
Please go https://github.com/ianrichr/jdk/actions and enable testing for your personal fork.

Then please update copyright in the file to:

```
 * Copyright (c) 2018, 2023, Oracle and/or its affiliates. All rights reserved.
```

...and push another commit here. This will trigger testing too.

File: test/jdk/sun/security/util/math/TestIntegerModuloP.java (hunk 1 above)
Comment (ianrichr) [reply test/jdk/sun/security/util/math/TestIntegerModuloP.java]: Updated with Copyright and removed the excess line. Please see latest commit.

**Comment by shipilev** [comment]
Hi @robilad, could you help us with OCA verification here a bit? Thanks!

**Comment by ianrichr** [comment]
Hi @wangweij and @XueleiFan, would either of you be willing to take a look at this PR? 😄

**Comment by shipilev** [comment]
> Thanks for bringing this to my attention. If you believe the test change will make it run smoother, just do it.

Would you like to formally approve this PR? Thanks!

**Comment by ianrichr** [comment]
@shipilev can you sponsor this? 😄

**Pull request closed without merging**